scripts/test-e2e-kubernetes.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# test-e2e-kubernetes.sh
+#
+# SUMMARY
+#
+#   Run E2E tests for Kubernetes.
+
+random-string() {
+  local CHARS="abcdefghijklmnopqrstuvwxyz0123456789"
+  # shellcheck disable=SC2034
+  for i in {1..8}; do
+    echo -n "${CHARS:RANDOM%${#CHARS}:1}"
+  done
+  echo
+}
+
+# Build a docker image if it wasn't provided.
+if [[ -z "${CONTAINER_IMAGE:-}" ]]; then
+  # Require a repo to put the container image at.
+  #
+  # Hint #1: you can use `scripts/start-docker-registry.sh`, but it requires
+  # manually preparing the environment to allow insecure registries, and it can
+  # also not work if you k8s cluster doesn't have network connectivity to the
+  # registry.
+  #
+  # Hint #2: if using with minikube, set `USE_MINIKUBE_CACHE` to `true` and use
+  # any value for `CONTAINER_IMAGE_REPO` (for instance, `vector-test` will do).
+  #
+  CONTAINER_IMAGE_REPO="${CONTAINER_IMAGE_REPO:?"You have to specify CONTAINER_IMAGE_REPO to upload the test image to."}"
+
+  # Assign a default test run ID if none is provided by the user.
+  TEST_RUN_ID="${TEST_RUN_ID:-"$(date +%s)-$(random-string)"}"
+
+  if [[ "${QUICK_BUILD:-"false"}" == "true" ]]; then
+    # Build in debug mode.
+    cargo build
```

Comment:
You should build vector correctly with the makefile here.

Reply:
I this case, the idea is to use the same build flow that devs usually use to reuse as much of the flow as possible. I'm using `cargo check` / `cargo test` usually, and not `make`. Thus `cargo build` if preferable to `make` here. Does it make sense?

Reply:
Oh, and there's a non-quick-build that uses makefile below. And that one is the primary one (used at CI), while this is an optional for quicker iteration.